connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/PostgresIntegrationSuite.scala

```diff
@@ -302,4 +317,22 @@ class PostgresIntegrationSuite extends DockerJDBCIntegrationV2Suite with V2JDBCT
     assert(rows10(0).getString(0) === "amy")
     assert(rows10(1).getString(0) === "alex")
   }
+
+  test("Test reading 2d array from table created via CTAS command") {
+    val dfWithNewBehaviour = sql(s"SELECT * FROM $catalogName.array_of_array_of_int")
+    val CTASdfWithNewBehaviour = sql(s"SELECT * FROM $catalogName.ctas_array_of_array_of_int")
+
+    checkAnswer(CTASdfWithNewBehaviour, dfWithNewBehaviour.collect())
+  }
+
+  test("Test reading multiple dimension array from table created via CTAS command") {
+    val exception = intercept[org.apache.spark.SparkSQLException] {
```

Comment:
Can we use `checkError` here? BTW, we just added this test in passing, not because of a behavioural change I guess?

Reply:
Done